### DISCUSSION TITLE
How can I change the backend of my admin account?

### DISCUSSION THREAD

**Discussion by poperigby** [mealie-recipes/mealie#4055] [Q&A] [answered]
I set it up as a local account during installation, but I actually want it to use OAuth so I can sign-in with Authelia.

**Comment by cmintey** [comment]
As long as your email in Authelia matches what is in Mealie, your sign in via OAuth will automatically link to your existing account.

**Reply by poperigby** [reply]
When I try to login with Authelia, it gives me this error in Mealie:

```
Aug 19 22:16:05 haddock mealie[4019802]: sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "ix_users_username"
Aug 19 22:16:05 haddock mealie[4019802]: DETAIL:  Key (username)=(admin) already exists.
Aug 19 22:16:05 haddock mealie[4019802]: [SQL: INSERT INTO users (id, full_name, username, email, password, auth_method, admin, advanced, group_id, cache_key, login_attemps, locked_at, can_manage, can_invite, can_organize, owned_recipes_id, created_at, update_at) VALUES (%(id)s::UUID, %(full_name)s, %(username)s, %(email)s, %(password)s, %(auth_method)s, %(admin)s, %(advanced)s, %(group_id)s::UUID, %(cache_key)s, %(login_attemps)s, %(locked_at)s, %(can_manage)s, %(can_invite)s, %(can_organize)s, %(owned_recipes_id)s::UUID, %(created_at)s, %(update_at)s)]
Aug 19 22:16:05 haddock mealie[4019802]: [parameters: {'id': 'e5fa7654-b3ab-408c-bf0c-5b4bdc938bf5', 'full_name': 'Administrator', 'username': 'admin', 'email': 'admin@haddock.cc', 'password': 'OIDC', 'auth_method': 'OIDC', 'admin': True, 'advanced': True, 'group_id': '0b48eb4d-e6f4-4d7b-aeff-e1de6dc51142', 'cache_key': '1234', 'login_attemps': 0, 'locked_at': None, 'can_manage': True, 'can_invite': True, 'can_organize': True, 'owned_recipes_id': None, 'created_at': datetime.datetime(2024, 8, 20, 5, 16, 5, 832922, tzinfo=datetime.timezone.utc), 'update_at': datetime.datetime(2024, 8, 20, 5, 16, 5, 832925, tzinfo=datetime.timezone.utc)}]
```

**Reply by cmintey** [reply]
Well it's not finding your user in Mealie with the email address and so trying to create a new one. Does your Mealie user have the same email address as your Authelia user? If not it seems like your usernames match, so you could add `OIDC_USER_CLAIM=preferred_username` to instead find a matching user by username.

You can also add `LOG_LEVEL=DEBUG` to your environment variables to get some more log messages to help debug.

**Reply by poperigby** [reply]
I think I might have somehow screwed up and created multiple admin accounts, because it said `admin` was already taken when I was doing the setup process and creating an admin. How can I go into the database and delete one of them?

**Reply by cmintey** [reply]
Do you still have access to one of those admin account? If yes, then you can go to this URL: `/admin/manage/users` and delete the user from the UI.
Otherwise accessing the database will vary depending on if you have sqlite or postgres. You can probably find resources in other discussions or by googling

**Reply by poperigby** [reply]
Unfortunately I don't have access to the admin. I have postgres, so should I just find the users table?

**Reply by cmintey** [reply, marked as answer]
Yeah, I believe the table name is just `users`. You can delete the other user there and also set your current user as admin

**Reply by poperigby** [reply]
Thanks! Looks like the admin user had the wrong email, because I forgot that restoring a backup wiped out the database.
